exercises/secret-handshake/test/Tests.hs

```diff
@@ -1,52 +1,51 @@
-main = exitProperly $ runTestTT $ TestList
-       [ TestList handshakeTests ]
-
--- The trick to this one is defining the appropriate typeclass or typeclasses
--- to make handshake work with both Int and String. Granted, this is probably
--- not good API design for this exercise, but the technique is useful.
--- You may want to see http://www.haskell.org/haskellwiki/List_instance
-
-handshakeTests :: [Test]
-handshakeTests =
-  [ testCase "1 to wink" $ do
-    ["wink"] @=? handshake (1::Int)
-    ["wink"] @=? handshake "1"
-  , testCase "10 to double blink" $ do
-    ["double blink"] @=? handshake (2::Int)
-    ["double blink"] @=? handshake "10"
-  , testCase "100 to close your eyes" $ do
-    ["close your eyes"] @=? handshake (4::Int)
-    ["close your eyes"] @=? handshake "100"
-  , testCase "1000 to jump" $ do
-    ["jump"] @=? handshake (8::Int)
-    ["jump"] @=? handshake "1000"
-  , testCase "11 to wink and double blink" $ do
-    ["wink", "double blink"] @=? handshake (3::Int)
-    ["wink", "double blink"] @=? handshake "11"
-  , testCase "10011 to double blink and wink" $ do
-    ["double blink", "wink"] @=? handshake (19::Int)
-    ["double blink", "wink"] @=? handshake "10011"
-  , testCase "11111 to jump, close your eyes, double blink, and wink" $ do
-    ["jump", "close your eyes", "double blink", "wink"] @=? handshake (31::Int)
-    ["jump", "close your eyes", "double blink", "wink"] @=? handshake "11111"
-  , testCase "zero" $ do
-    [] @=? handshake (0::Int)
-    [] @=? handshake "0"
-  , testCase "gibberish" $
-    [] @=? handshake "piggies"
-  , testCase "partial gibberish" $
-    [] @=? handshake "1piggies"
-  ]
+    it "1 to wink" $ do
+      handshake (1 :: Int) `shouldBe` ["wink"]
+      handshake "1"        `shouldBe` ["wink"]
+
+    it "10 to double blink" $ do
+      handshake (2 :: Int) `shouldBe` ["double blink"]
+      handshake "10"       `shouldBe` ["double blink"]
+
+    it "100 to close your eyes" $ do
+      handshake (4 :: Int) `shouldBe` ["close your eyes"]
+      handshake "100"      `shouldBe` ["close your eyes"]
+
+    it "1000 to jump" $ do
+      handshake (8 :: Int) `shouldBe` ["jump"]
+      handshake "1000"     `shouldBe` ["jump"]
+
+    it "11 to wink and double blink" $ do
+      handshake (3 :: Int) `shouldBe` ["wink", "double blink"]
+      handshake "11"       `shouldBe` ["wink", "double blink"]
+
+    it "10011 to double blink and wink" $ do
+      handshake (19 :: Int) `shouldBe` ["double blink", "wink"]
+      handshake "10011"     `shouldBe` ["double blink", "wink"]
+
+    it "11111 to jump, close your eyes, double blink, and wink" $ do
+      handshake (31 :: Int) `shouldBe` ["jump", "close your eyes", "double blink", "wink"]
+      handshake "11111"     `shouldBe` ["jump", "close your eyes", "double blink", "wink"]
+
+    it "zero" $ do
+      handshake (0 :: Int) `shouldBe` []
+      handshake "0"        `shouldBe` []
+
+    it "gibberish" $
+      handshake "piggies" `shouldBe` []
```

Comment:
I was rewriting this PR to deal with `Maybe` values - which makes sense - but I noticed while rewriting the example solution that everything felt so artificial with this exercise. Even the test suite makes it clear that the API design is far from good:

> The trick to this one is defining the appropriate typeclass or typeclasses
> to make handshake work with both Int and String. Granted, this is probably
> not good API design for this exercise, but the technique is useful.

I'm convinced that this exercise deserves a complete rewrite, so I propose:

1. Merge this PR the way it is, just to keep moving the migration to `hspec`.
2. Wait for exercism/x-common#159 to merge, luckily with a more sane set of tests.
3. Completely rewrite this exercise from scratch.